game/src/contrib/systems/HealthSystem.java

```diff
@@ -29,45 +29,45 @@ public HealthSystem() {
     public void execute() {
         entityStream()
                 // Consider only entities that have a HealthComponent
-                // Form triples (e, hc, ac)
+                // Form triples (e, hc, dc)
                 .map(this::buildDataObject)
                 // Apply damage
                 .map(this::applyDamage)
                 // Filter all dead entities
                 .filter(hsd -> hsd.hc.isDead())
                 // Set DeathAnimation if possible and not yet set
-                .map(this::triggerDeathAnimation)
-                .filter(this::filterByAnimation)
+                .map(this::activateDeathAnimation)
+                .filter(this::testDeathAnimationStatus)
                 // Remove all dead entities
                 .forEach(this::removeDeadEntities);
     }
 
     /**
-     * Filters the stream by checking the Animation of an Entity.
+     * Tests the existence and current status of the DeathAnimation of an Entity. Returns a
+     * corresponding boolean if the Entity can be removed from the game.
      *
      * @param hsd HSData to check Animations in
      * @return true if Entity can be removed from the game
      */
-    private boolean filterByAnimation(HSData hsd) {
+    private boolean testDeathAnimationStatus(HSData hsd) {
+        DrawComponent dc = hsd.dc;
         // test if hsd has a DeathAnimation
-        Predicate<HSData> hasDeathAnimation =
-                (hsData) -> hsData.ac.hasAnimation(AdditionalAnimations.DIE);
+        Predicate<DrawComponent> hasDeathAnimation =
+                (drawComponent) -> drawComponent.hasAnimation(AdditionalAnimations.DIE);
         // test if Animation is looping
-        Predicate<HSData> isAnimationLooping =
-                (hsData) -> hsData.ac.getAnimation(AdditionalAnimations.DIE).get().isLooping();
+        Predicate<DrawComponent> isAnimationLooping = DrawComponent::isCurrentAnimationLooping;
         // test if Animation has finished playing
-        Predicate<HSData> isAnimationFinished =
-                (hsData) -> hsData.ac.currentAnimation().isFinished();
+        Predicate<DrawComponent> isAnimationFinished = DrawComponent::isCurrentAnimationFinished;
 
-        return !hasDeathAnimation.test(hsd)
-                || isAnimationLooping.test(hsd)
-                || isAnimationFinished.test(hsd);
+        return !hasDeathAnimation.test(dc)
+                || isAnimationLooping.test(dc)
```

Comment:
müsste das hier nicht `|| !isAnimationLooping.test(dc)` sein?

die entität kann entfernt werden: (a) sie hat keine sterbe-animation, oder (b) die animation ist nicht mehr am loopen, oder (c) sie ist beendet?

oder hab ich das falsch verstanden mit den animationen?

Reply:
hab noch mal bei den anderen nachgefragt
der Check auf isAnimationLooping ist dazu da, um Entities mit loopenden DeathAnimationen rauszufiltern, weil die sonst immer wieder umknicken, aber nie aus dem Spiel entfernt werden

ich hab den Test jetzt mal vorgezogen und ein warning log mit hinzugefügt, dadurch ist die Funktionalität beibehalten, aber der User bekommt mit was falsch läuft

und dass die Animation nicht loopt ist zusätzlich noch in isAnimationFinished eine Bedingung
die Animation kann nur Finished sein, wenn sie explizit nicht loopt, da brauchen wir keinen doppelt gemoppelten check :smile: